openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,554 @@
+        resolution : int
+            Value used to calculate the size of contour plot meshgrid
+        """
+        self.prob = prob
+        self.surrogate_comp = surrogate_comp
+        self.resolution = resolution
+        # Create list of inputs
+        self.input_list = [i[0] for i in self.surrogate_comp._surrogate_input_names]
+
+        if len(self.input_list) < 2:
+            raise ValueError('Must have more than one input value')
+
+        self.output_list = [i[0] for i in self.surrogate_comp._surrogate_output_names]
+
+        # Pair input list names with their respective data
+        self.input_data = {}
+        for title in self.input_list:
+            try:
+                self.input_data[title] = {
+                    i for i in self.surrogate_comp.options[str('train:' + title)]}
+            except TypeError:
+                msg = "No training data present for one or more parameters"
+                raise TypeError(msg)
+
+        # Setup dropdown menus for x/y inputs and the output value
+        self.x_input = Select(title="X Input:", value=[x for x in self.input_list][0],
+                              options=[x for x in self.input_list])
+        self.x_input.on_change('value', self._x_input_update)
+
+        self.y_input = Select(title="Y Input:", value=[x for x in self.input_list][1],
+                              options=[x for x in self.input_list])
+        self.y_input.on_change('value', self._y_input_update)
+
+        self.output_select = Select(title="Output:", value=[x for x in self.output_list][0],
+                                    options=[x for x in self.output_list])
+        self.output_select.on_change('value', self._output_value_update)
+
+        # Create sliders in a loop
+        self.slider_dict = {}
+        self.input_data_dict = OrderedDict()
+        for title, values in self.input_data.items():
+            slider_data = np.linspace(min(values), max(values), self.resolution)
+            self.input_data_dict[title] = slider_data
+            # Calculates the distance between slider ticks
+            slider_step = slider_data[1] - slider_data[0]
+            self.slider_dict[title] = Slider(start=min(values), end=max(values), value=min(values),
+                                             step=slider_step, title=str(title))
+
+        # Match the slider dictionary key value pairs with an on
+        # change event handler to call an update function later
+        for name, slider_object in self.slider_dict.items():
+            if name == self.x_input.value:
+                self.x_input_slider = slider_object
+                self.x_input_slider.on_change('value', self._scatter_plots_update)
+            elif name == self.y_input.value:
+                self.y_input_slider = slider_object
+                self.y_input_slider.on_change('value', self._scatter_plots_update)
+            else:
+                setattr(self, name, slider_object)
+                obj = getattr(self, name)
```

Comment:
It's not clear to me why you are creating these attributes, rather than just doing `slider_object.on_change(...)`
I don't see where they are used?

Reply:
I create these attributes because I create a slider for every input so that a user can reduce an n dimensional input space to 3 dimensions. Consider a 4 dimensional space, the user will select 2 dimensions to be a single value input while the remaining 2 dimension input sliders allow the user to make slices in the contour at any given point which are then projected on to the subplots. 

I create them so that each slider has a unique name in the slider_dict which will later be looped through so bokeh can display them

Reply:
Upon further clarification and discussion, I can just use `slider_object.on_change()`